### PR TITLE
Change ppc64 to use correct `POWERPC` target

### PR DESCRIPTION
**Pull request by Grounded0**
# Description

There's three different Power ISA platforms: powerpc (PowerPC 32-bit), ppc64 (PowerPC 64-bit), and ppc64le  (PowerPC 64-bit little-endian).

Changing target from incorrect `PPC64LE` to correct `POWERPC` for ppc64 platform.

## Related issues

- #3145

# Manual testing

No hardware to test on.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by classilla** [comment]
Shouldn't break anything, but I still haven't heard if the JIT even works properly on big-endian PowerPC64.

**Comment by johnnovak** [comment]
> Shouldn't break anything, but I still haven't heard if the JIT even works properly on big-endian PowerPC64.

Good point. On the other hand, there is no hope in hell for the LE build to even start up on BE PPC, am I right? So with the correct LE build people could at least use the non-dynarec cores.

Also, ignore the Windows breakages; those were me building 0.81.2 and forgetting to switch back the Windows workflows.

**Comment by Grounded0** [comment]
PowerPC 64-bit Kernel Internals

https://lwn.net/2001/features/OLS/pdf/pdf/ppc64.pdf

> IBM’s 64-bit implementations of the PowerPC processor architecture provide full 32-bit execution compatibility with no performance loss, a feature that allows the Linux/PPC64 to provide excellent support of existing Linux/PPC32 applications.

**Comment by Grounded0** [comment]
> Good point. On the other hand, there is no hope in hell for the LE build to even start up on BE PPC, am I right? So with the correct LE build people could at least use the non-dynarec cores.

Yes.

**Comment by johnnovak** [comment]
> > Good point. On the other hand, there is no hope in hell for the LE build to even start up on BE PPC, am I right? So with the correct LE build people could at least use the non-dynarec cores.
> 
> Yes. With the correct BE build.

Ah, yes, fixed in my original comment.

**Comment by Grounded0** [comment]
@casey-ac plays Theme Hospital with Power Mac G5 so I would say dynarec definitely works:

- #3144

File: src/cpu/meson.build
Comment (runlevel5) [inline src/cpu/meson.build:26]: Can't we duplicate POWERPC to `PPC64` or `PPC64BE` to separate the 2 targets?

Comment (johnnovak) [reply src/cpu/meson.build:26]: I think if you want more substantial changes, you'll need to do the work yourself as a PPC user 😄

Comment (classilla) [reply src/cpu/meson.build:26]: Actually, it just dawned on me what's going on. This is in fact specifying the _32-bit_ PowerPC JIT, which is inherently big-endian. But 32-bit instructions will run fine on a 970MP. This is probably the best option for now since I'm still pretty sure the 64-bit JIT isn't necessarily endian-safe, barring any of the weird edge cases we've smoked out of the 64-bit Power dynrec.

**Review by johnnovak — APPROVED** [review]
Looks fine to me. Any other substantial PPC-related changes need to be a community effort as the core team has neither the means nor the interest to test Staging on PPC.

**Comment by Grounded0** [comment]
@johnnovak 

Since I'm a maintainer is this "Rebase and merge" option safe or am I going to blow up something?

**Comment by johnnovak** [comment]
> @johnnovak
> 
> Since I'm a maintainer is this "Rebase and merge" button safe to press or am I going to blow up something?

I've just pressed rebase. Let it run the CI workflows again, then let's merge once all green (feel to press the merge button).

**Pull request merged**
